### ISSUE TITLE
Fix pubsub.asyncIterator typescript type

### ISSUE THREAD

**Issue by xialvjun** [apollographql/apollo-server#4556]
package: graphql-subscriptions
version: 1.1.0

expected behavior:  code like down is typescript ok
```ts
Subscription: {
  on_new_count: {
    subscribe: async function*(_, args, ctx, info) {
      const iter = pubsub.asyncIterator<number>('on_new_count');
      // this line errors
      for await (const iterator of iter) {
      }
    },
    resolve: (_: number) => _,
  },
};
```

actual behavior: typescript errors `Type 'AsyncIterator<number, any, undefined>' must have a '[Symbol.asyncIterator]()' method that returns an async iterator.`

**Comment by xialvjun** [comment]
I know I can use `withFilter`, but then type generated by `graphql-codegen` will be lost , ex - args should can be its real type, but using `withFilter`, I got `any`

**Comment by sasa6277** [comment]
i solve it with  subscribe: async (
      _: void,
      { Inputs: { chatId } }: { Inputs: SendMessageSubVariable }
    ): Promise<AsyncIterator<unknown, { message: string }, undefined>> => {
      const key = messageKey(chatId);
      return pubsub.asyncIterator([key]);
    }
  }

**Comment by Miljoen** [comment]
Hi guys I came across this issue and this worked for me:

```
    Subscription: {
        userUpdated: {
            subscribe: () => ({
                [Symbol.asyncIterator]: () => pubSub.asyncIterator(['USER_UPDATED']),
            }),
        },
    },
```

Basically the best way to handle this is to make your `Iterator` an `Iterable`

Issue came up as part of strongly typing my Apollo Server Resolvers using Codegen

**Comment by glasser** [comment]
Looks like this is more relevant to the (not super actively maintained, unfortunately) `graphql-subscriptions` project rather than `apollo-server` so I will close this issue.  Looks like some related discussion on https://github.com/apollographql/graphql-subscriptions/issues/143 and https://github.com/apollographql/graphql-subscriptions/issues/192

Actually I think this might be fixed by https://github.com/apollographql/graphql-subscriptions/pull/232 which @hwillson has scheduled to be released in 3.0 https://github.com/apollographql/graphql-subscriptions/pull/250
